pkg/apis/autoscaling/validation/validation.go

```diff
@@ -197,6 +197,10 @@ func validateExternalSource(src *autoscaling.ExternalMetricSource, fldPath *fiel
 		allErrs = append(allErrs, field.Required(fldPath.Child("metricName"), "must specify a metric name"))
 	}
 
+	for _, msg := range pathvalidation.IsValidPathSegmentName(src.MetricName) {
```

Comment:
can put this in an else block if `len(src.MetricName) > 0`

Reply:
done